utils/registry/auth/handler.go

```diff
@@ -103,8 +131,10 @@ func (t *standardTokenHandler) AuthorizeRequest(req *http.Request, params map[st
 		return err
 	}
 
-	// TODO support secretKey
-	r.SetBasicAuth(t.credential.Username, t.credential.Password)
+	if t.credential.Type() == basicAuth {
```

Comment:
use a func such as PrepareReq and there won't need such if, the code would scale better?